framework/core/src/Post/Access/PostPolicy.php

```diff
@@ -71,6 +71,14 @@ public function edit(User $actor, Post $post)
      */
     public function hide(User $actor, Post $post)
     {
-        return $this->edit($actor, $post);
+        if ($post->user_id == $actor->id && (! $post->hidden_at || $post->hidden_user_id == $actor->id) && $actor->can('reply', $post->discussion)) {
```

Comment:
I wonder if we should drop the bit about only being able to hide when the discussion is still open 🤔 
```suggestion
        if ($post->user_id == $actor->id && (! $post->hidden_at || $post->hidden_user_id == $actor->id)) {
```

Reply:
I tried to keep the behavior close to the post-editing one for consistency so that's why that's there! I was wondering if it would be necessary or not.

Reply:
Yeah, not sure, let's keep it as is for now, then we'll see I guess unless other core devs have an opinion on is.

Reply:
I'd prefer to keep it. If a user can't participate in a discussion going forward, i don't think they should be able to participate retroactively either.